### ISSUE TITLE
[BUG] clang: compiling a simple function for the bpf target produces invalid output

### ISSUE THREAD

**Issue by lmb** [compiler-explorer/compiler-explorer#2674]
**Describe the bug**
Some functions have inaccurate assembly output when compiling using clang for the bpf target.

```C
int foo() {
    return sizeof(int);
}
```

The load of `sizeof(int)` into a register is missing in the assembly output.

**Expected behavior**

Compiling the same code locally against clang-12 yields this:

```
0000000000000000 <foo>:
llvm-objdump: warning: 'min.o': failed to parse debug information for min.o
       0:	b7 00 00 00 04 00 00 00	r0 = 4
       1:	95 00 00 00 00 00 00 00	exit
```

Note the presence of `r0 = 4`.

**Link**
https://godbolt.org/z/4E1s3EKeq

**Comment by partouf** [comment]
I fear this method of notation isn't easy to support because it's usually assumed with assignment notation that labels are involved, where here a register is involved. But this is a good example of which we can base unittests on.

Thanks for reporting this.

**Comment by OfekShilon** [comment]
This is now fixed, probably by #5706. @lmb can you please verify, and if indeed fixed - close?
